### PR TITLE
Revise Project page and StaffPlan pages UI

### LINKED ISSUE #369: Revise UI location for adding (+) projects
No more `+` pluses everywhere for adding a project.
The `+` is removed from the Client and ClientName text.
The `+` is now being moved to the left of the `Project` column header text.
Only the `+` is clickable to launch the adding of a project.

On click of `+`, let's insert the UI input fields for a new project (maybe quickly animating in) at the top of the list of projects so it's close to where I just clicked (instead of the bottom).
See related ticket: https://github.com/orgs/goinvo/projects/2/views/1?pane=issue&itemId=89735024&issue=goinvo%7Cstaffplan-next-app%7C399

And upon clicking the `+` my focus should be placed inside the Client Name input field so I can quickly type. And then tabbing should bring me to the Project Name input field. And then tabbing out or hitting enter should confirm/solidify that project to feel like the rest of the existing projects.

Note: 
The `+` doesn't currently function on Projects and My StaffPlan.
On My StaffPlan, the new project input fields are displayed by default, we should remove those unless the person clicks the `+`.

add-project

### LINKED ISSUE #371: Left align left-sided content and vertically align project row content
From the nav to all the text running along the left side, let's line them all up.
Across all views.

Notice that there's only one + button... it's moved and not repeated with clients. (also mentioned as separate ticket).

left-align-text

And also vertically align the client, project name, proposed/signed.
vertical-alignment

### LINKED ISSUE #399: Revise new project interaction on My StaffPlan or any Person’s StaffPlan view.
Looking to revise the layout and interaction for adding a new project on My StaffPlan.

- Remove the open input fields at the bottom client name and project name. These will not display by default, but must be triggered by a click.
- Upon clicking the `+` link beside `Project`, a project row will be inserted at the top of the list of projects. Can we do a quick animation to expand that row for the new project inputs so it’s not so jarring? 
- Upon displaying the new project row, place focus on the client name input so the person can quickly enter in the details without using their cursor.
- Hitting Enter will try to submit the input fields and save the project. Hitting Save button will do the same.
- If successful, the new project will swap from the input fields into the normal view of a project, showing the week hour input fields, and as a proposed project. Can this be a nice, quick, subtle fade-out/in transition to that new view?
- Do not automatically re-sort the project upon added. The shuffling makes it hard to follow where it went while I’m trying to focus on it. Only re0sort on next refresh of that screen.
- If I’ve opened that new project input fields row and then refresh the screen, that row is removed. Said another way, an incomplete new project should not persist.
- Change placeholder text for Client Name to `Client`. This will be consistent with how it's described throughout rest of the service.
- Note: Trying to line up the input fields and their placeholder text for a new project with the existing projects below it.
- Note: Are we presenting an error if the project name is already in use? Should we allow it and just display it on that persons My StaffPlan view? I think we should allow them to add an existing project.

new-project-flow-from-mystaffplan

### PR DESCRIPTION
**Pull request by zhukovdigital**
closes #369 
closes #399 
closes #371

### CODE REVIEW

**Comment by zhukovdigital** [comment]
closes #460

**Comment by HankC138** [comment]
@zhukovdigital 
failed to deploy because of a missing type here is a screenshot
image

**Review by HankC138 — APPROVED** [review]
